src/operators.jl

```diff
@@ -442,3 +442,43 @@ This is equivalent to [`induced_subgraph`](@ref)`(g, neighborhood(g, v, d, dir=d
 with respect to `v` (i.e. `:in` or `:out`).
 """
 egonet(g::AbstractGraph, v::Integer, d::Integer; dir=:out) = g[neighborhood(g, v, d, dir=dir)]
+
+
+"""
+    merge_vertices!(g::Graph, vs::Array{Int})
+
+Combines vertices specified in `vs` into single vertex whose
+index will be the lowest value in `vs`.
+
+All edges running to vertices in `vs` will connect to new
+merged vertex.
+
+Returns an Array{Int} with new vertex values for
+"""
+
+function merge_vertices!(g::Graph, vs::Array{Int})
+
+    allunique(vs) || error("Vertices in vs must be unique")
+
+    vertex_conversions = collect(1:nv(g))
+
+    # Use lowest value as new vertex id.
+    v0 = minimum(vs)
+    filter!(x->x != v0, vs)
```

Comment:
Does this require length(vs) operations to remove the one element?
When you do `allunique` it uses a Set to determine the unique elements.
You could instead use `sort!(vs)` and then use `vs = vs[2:end]`